torchaudio/sox_effects/sox_effects.py

```diff
@@ -339,6 +339,11 @@ def __init__(self,
                  out_siginfo: Any = None,
                  out_encinfo: Any = None,
                  filetype: str = "raw") -> None:
+        warnings.warn(
+            'torchaudio.sox_effects.SoxEffectsChain has been deprecated and '
+            'will be removed from 0.8.0 release. '
+            'Please migrate to `apply_effects_file` or `apply_effects_tensor`.'
+        )
```

Comment:
maybe make it filterable?
```suggestion
            'Please migrate to `apply_effects_file` or `apply_effects_tensor`.',
            DeprecationWarning
        )
```

Reply:
Thanks for suggestion. Although I agree with you and that is better, but somehow domain libraries have rules to use `UserWarning` for deprecation.

Reply:
Ok, sorry for the noise (again)

Reply:
No worries, I appreciate your feedback. :)
I think we should sit together to sync up on your documentation work soon.

Reply:
OK. Maybe we can bet @brianjo involved.

Reply:
> Thanks for suggestion. Although I agree with you and that is better, but somehow domain libraries have rules to use `UserWarning` for deprecation.

This is also standard with pytorch. The issue with DeprecationWarning is that it is hidden by default.